### ISSUE TITLE
Internal seed of keras.layers.Dropout gets increased by +1 every time a tf.function gets created from the model

### ISSUE THREAD

**Issue by mergian** [keras-team/tf-keras#60]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux CentOS
- TensorFlow installed from (source or binary): PYPI
- TensorFlow version (use command below): 2.11.0
- Python version: 3.7
- Bazel version (if compiling from source):
- GPU model and memory:
- Exact command to reproduce:

**Describe the problem**.

When using `keras.layers.Dropout` with a user defined seed, the seed gets increments by `+1` every time `tf.function` gets called (i.e. in `make_train_function`, ...).

See code and log below. As you can see, every time I run `tf.function`, the seed of the Keras dropout layer gets increased. BUT the seed of manually crafted Dropout using `tf.random.uniform` does not. So I assume it's a Keras problem. I tried to pinpoint the error and found the following code:

https://github.com/keras-team/keras/blob/d2a6b9e0efdd01dd1bc6cf0bb52336d0009aba6c/keras/backend.py#L2007-L2025

Here the seed gets increased "When user didn't provide any original seed". However, the check is wrong, as `self._seed` is the user defined seed. It get set in the constructor and also here: https://github.com/keras-team/keras/blob/d2a6b9e0efdd01dd1bc6cf0bb52336d0009aba6c/keras/backend.py#L1974-L1976
https://github.com/keras-team/keras/blob/d2a6b9e0efdd01dd1bc6cf0bb52336d0009aba6c/keras/backend.py#L2027-L2033

So I think the `make_legacy_seed` function needs to be fixed. I'm not sure about this comment "... it is important to generate different seed for stateful ops ...", because the underlying `tf.random.uniform` stores the state in the `PhiloxRandom` C++ class, which gets incremented throughout the training process and only gets reset when triggered by `tf.random.set_seed(...)`. I don't see why the seed needs to be manually increased here.

**Describe the current behavior**.
Every time we run `tf.function` the seed of the `keras.layers.Dropout` gets increased.

**Describe the expected behavior**.
The seeds should be stable, so we can have deterministic behavior.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

Personally I would remove the `make_legacy_seed` method, but I'm not sure if there was a special purpose for this behavior. Alternative, we could store a `self.user_provided_seed = seed is not None` and check for this within `make_legacy_seed`.

**Standalone code to reproduce the issue**.

```python
import tensorflow as tf
import numpy as np

tf.random.set_seed(123)

rate = 0.5
inp = tf.keras.Input((5,))
x = tf.keras.layers.Dropout(rate=rate, seed=345)(inp)
y = tf.cast(tf.random.uniform(shape=tf.shape(inp), dtype=inp.dtype, seed=345) >= rate, inp.dtype) * inp * (1/rate)
model = tf.keras.Model(inputs=[inp], outputs=[x, y])

args = np.ones((1, 5,), np.float32)
counter = 0

def print_seeds():
    global counter
    print(f"Run #{counter}")
    counter += 1

    print("\tModel:")
    for l in model.layers:
        if hasattr(l, 'seed'):
            print("\t\t", l.name, l.seed)

    print("\tGraph:")
    func = tf.function(model.__call__)
    graph = func.get_concrete_function(args, training=True).graph
    for node in graph.get_operations():
        if node.op_def.name == "RandomUniform":
            print("\t\t", node.name, 'Seed:', node.get_attr('seed'), 'Seed2:', node.get_attr('seed2'))
    print()

print_seeds()
print_seeds()
print_seeds()
```

**Source code / logs**.

```
Run #0
        Model:
                 dropout 345
        Graph:
                 model/tf.random.uniform/random_uniform/RandomUniform Seed: 123 Seed2: 345
                 model/dropout/dropout/random_uniform/RandomUniform Seed: 123 Seed2: 345

Run keras-team/keras#1
        Model:
                 dropout 345
        Graph:
                 model/tf.random.uniform/random_uniform/RandomUniform Seed: 123 Seed2: 345
                 model/dropout/dropout/random_uniform/RandomUniform Seed: 123 Seed2: 346

Run keras-team/keras#2
        Model:
                 dropout 345
        Graph:
                 model/tf.random.uniform/random_uniform/RandomUniform Seed: 123 Seed2: 345
                 model/dropout/dropout/random_uniform/RandomUniform Seed: 123 Seed2: 347
```
